### PR TITLE
Add a state machine library

### PR DESCRIPTION
**Pull request by just-now**

### CODE REVIEW

File: src/utils.h [hunk 1]
```diff
@@ -13,4 +14,13 @@
 
 #define DBG() fprintf(stderr, "%s:%d\n", __func__, __LINE__)
 
+#define UNUSED __attribute__((unused))
+
+#define container_of(e, type, field) \
+	((type *)(uintptr_t)((char *)(e)-offsetof(type, field)))
```
Comment (cole-miller) [inline src/utils.h:20]: I slightly prefer CONTAINER_OF for function-like macros

File: src/utils.h [hunk 2]
```diff
@@ -13,4 +14,13 @@
 
 #define DBG() fprintf(stderr, "%s:%d\n", __func__, __LINE__)
 
+#define UNUSED __attribute__((unused))
+
```
Comment (cole-miller) [inline src/utils.h:18]: Let's use casting to `void` for this purpose instead

**Review by cole-miller — CHANGES_REQUESTED** [review]
Only stylistic comments

File: src/utils.h (hunk 1 above)
Comment (just-now) [reply src/utils.h:20]: I have fixed this.

Still i'd notice that it's well known macro people used to write in lowercase. No one would treat it as a function.

File: src/utils.h (hunk 2 above)
Comment (just-now) [reply src/utils.h:18]: fixed.

**Pull request merged**